### PR TITLE
feat(core+cli): ADR-091 Stage 4 baseline overrides + matchesGlob consolidation + manifest self-match exclusion

### LINKED ISSUE #1683: core+cli: ADR-091 Stage 4 verification baseline semantics + config field
## Context

Companion ticket to mmnto-ai/totem#1682 (Stage 4 verifier headline). Defines the **verification baseline** — the file set the verifier expects compiled rules NOT to fire on. Per [ADR-091 Deferred Decisions §"Verification baseline semantics for Stage 4"](https://github.com/mmnto-ai/totem-strategy/blob/main/adr/adr-091-ingestion-pipeline-refinements.md).

**Naming-note enforcement.** "Verification baseline" is the canonical term, **not** "allowlist." An earlier ADR-091 draft used `allowlist`, which GCA review flagged as counter-intuitive in lint/security parlance where `allowlist` usually means "exempt from flagging." The verification baseline is the inverse: files the rule is required to NOT fire on. Schema field names, config field names, doc copy, and CLI surface text must all use "baseline" framing.

### Default baseline rules

- [ ] **Outside `fileGlobs`** — files outside the lesson's declared `fileGlobs` scope are implicitly in the baseline. A rule scoped to `**/*.test.ts` that fires on `src/server.ts` is over-broad by definition.
- [ ] **Test files** — `**/*.test.*`, `**/*.spec.*`, `**/tests/**/*.*`, `**/__tests__/**/*.*` in the baseline by default. Override path: rules whose `fileGlobs` explicitly includes test patterns (the test-contract scope class shipped in #1652) are NOT implicitly excluded from their own scope.
- [ ] **Fixture directories** — `**/fixtures/**`, `**/__fixtures__/**` in the baseline by default. Same override path as test files.

### Consumer override surfaces

- [ ] **`.totemignore` extension** — explicit baseline-marker comment syntax (e.g., `# stage4-baseline: ...`) on top of the existing ignore semantics. Schema design open: prefer minimal divergence from the existing file format. Document the extension in `docs/wiki/cli-reference.md` `.totemignore` section.
- [ ] **`review.stage4Baseline` config field** on `totem.config.ts`. Type and shape:
  - `extend: string[]` — additional glob patterns added to the default baseline.
  - `exclude: string[]` — globs removed from the default baseline (e.g., a project that wants its `tests/` dir out of the baseline).
  - Zod-validated; fail-loud on schema violations.
- [ ] **Schema-name discipline** — every public surface uses "baseline" framing: TS field names, JSON config keys, error-message text, CLI flag names if any. No `allowlist` aliases. (Per the GCA finding logged in ADR-091 Deferred Decisions.)

### Wiring

- [ ] mmnto-ai/totem#1682 verifier consumes the resolved baseline through a single `resolveStage4Baseline(config, lesson)` helper exported from `@mmnto/totem`. The helper computes: defaults ∪ `extend` ∖ `exclude` ∖ files matching `lesson.fileGlobs`.
- [ ] Test coverage: unit tests for default baseline computation, `.totemignore` extension parsing, `review.stage4Baseline` schema validation (happy + sad paths), and the `extend`/`exclude` precedence rules.

## Dependencies

- **Parallel with mmnto-ai/totem#1682** — T1 ships with a stub default baseline if T2 has not landed yet. Once T2 lands, T1 swaps the stub for the `resolveStage4Baseline` helper.
- **Blocks no further tickets directly** — T3/T4/T5 depend on T1's verifier interface, which is independent of baseline semantics.

## Out of Scope

- Stage 4 verifier itself — see mmnto-ai/totem#1682.
- Per-rule baseline overrides (a single rule defining its own baseline outside the consumer-level config). Defer until a use case surfaces. The default-plus-consumer-override mechanism covers the known cases.

## References

- ADR-091 Deferred Decisions §"Verification baseline semantics for Stage 4" — `mmnto-ai/totem-strategy/blob/main/adr/adr-091-ingestion-pipeline-refinements.md`
- Decomposition audit doc — `mmnto-ai/totem-strategy/blob/main/audits/internal/2026-04-25-adr-091-decomposition.md`
- Existing `.totemignore` semantics — `docs/wiki/cli-reference.md`
- GCA "allowlist" naming finding — original PR #1607 review thread referenced in ADR-091 Deferred Decisions

### LINKED ISSUE #1758: Stage 4 verifier: rewrite matchesGlob with proper segment-aware matcher (CR mmnto-ai/totem#1757 R2)
## Context

Surfaced by CR R2 on `mmnto-ai/totem#1757` (Stage 4 substrate ship). The custom glob matcher in `packages/core/src/stage4-verifier.ts:146-196` (`matchesGlob` + `globToRegex`) has two known false-positive shapes:

1. **Substring overmatch on `**/segment/**` form.** `**/tests/**` lowers to the regex `^.*tests/.*$`, which matches `src/contests/foo.ts` (substring) instead of just `tests/`-anchored paths. Same issue on `**/__tests__/**` (would substring-match a path containing `notests` etc., though less likely in practice).

2. **Suffix overmatch on literal fallback.** The literal-glob branch checks `normalized === glob || normalized.endsWith('/' + glob)`. A glob like `src/foo.ts` matches `packages/src/foo.ts` via the suffix check — Stage 4 baseline classification can flip on filenames the consumer never authored.

## Why deferred from #1757

Existing code from `447e9065` (R0 substrate). My CR R2 patch addresses the 6 actionable findings + 2 critical Sonnet findings on the diff surface; swapping the glob matcher would:

- Expand the diff surface beyond the CR R2 turn.
- Touch a load-bearing dispatch (`fileMatchesGlobs` in `stage4-verifier.ts` is the gate that partitions violations into baseline vs. in-scope, and any change risks a four-outcome regression).
- Need a separate decomposition: pick a matcher (minimatch / picomatch / a hand-rolled segment-aware regex), audit the perf cost (T5 `mmnto-ai/totem#1686` already has the perf-pass scope), and confirm parity with `rule-engine.ts:fileMatchesGlobs` (which already has a trusted matcher used by `applyRulesToAdditions`).

## Acceptance criteria

- `matchesGlob` either reuses the existing `fileMatchesGlobs` matcher from `packages/core/src/rule-engine.ts` (preferred — single source of truth) or imports `picomatch` / `minimatch` and configures it for segment-aware matching.
- Both anti-cases get explicit regression tests in `stage4-verifier.test.ts`:
  - `**/tests/**` does NOT match `src/contests/foo.ts`.
  - Literal `src/foo.ts` does NOT match `packages/src/foo.ts`.
- `globToRegex` either gets removed or stays as a leaf helper if the new matcher needs a fallback.
- All four-outcome contract tests still pass — no behavioral regression on the canonical substrate cases (`#1526`-class regression must still fire `out-of-scope`).

## Pairing

Naturally rides with `mmnto-ai/totem#1683` (T2 — consumer baseline overrides via `review.stage4Baseline` config field). T2 already touches the glob layer to support `extend`/`exclude` overrides; bundling the matcher swap there means one re-test pass instead of two. Alternative: standalone PR if T2 is delayed.

## CR finding reference

Inline comment on `mmnto-ai/totem#1757` (review id `4203293219`): https://github.com/mmnto-ai/totem/pull/1757#discussion_r3166298582 (or the equivalent comment ID 3166298582).

> The custom glob matcher is not path-accurate. Two concrete misclassifications here: `**/tests/**` becomes `^.*tests/.*$`, so `src/contests/foo.ts` is treated as baseline, and the literal fallback makes `src/foo.ts` match `packages/src/foo.ts` via `endsWith('/' + glob)`. Both can flip a valid rule into `out-of-scope` or hide a real in-scope hit.

### LINKED ISSUE #1765: core: Stage 4 verifier self-matches every regex rule against `.totem/compiled-rules.json` manifest
## Mechanical root cause

`packages/core/src/stage4-verifier.ts:289` (`runRuleAgainstAllFiles`) intentionally strips `fileGlobs` so the rule fires on every file in the listed corpus, with the partition into baseline-vs-in-scope happening in `classifyFile` afterward. The comment at `stage4-verifier.ts:281-288` documents this as the "without the strip, baseline files would be skipped internally and we'd see zero out-of-scope hits even when the pattern is obviously over-broad" rationale.

Side effect: the rule fires on `.totem/compiled-rules.json` itself, because that file contains the rule's own `badExample` field as JSON data. The badExample is, by definition, a string the rule's regex pattern is supposed to match. So:

1. Rule pattern matches its own `badExample` line in compiled-rules.json.
2. `compiled-rules.json` doesn't match any source-scoped `fileGlobs` (rules typically scope to source code paths like `packages/**/*.rs`).
3. `classifyFile` returns `'baseline'` (file is outside `fileGlobs` → implicitly baseline).
4. Verifier short-circuits to `outcome: 'out-of-scope'` (line 425-431) — even one baseline match wins precedence.
5. Rule archives via `stage4-out-of-scope-match` reasonCode.

## Why this matters

**Every regex rule with a `badExample` field will Stage 4-archive on its first compile run, regardless of whether the rule is actually over-broad on real source code.**

Demonstrated on `mmnto-ai/totem#1761` AC #1 probe (LC's `lesson-639c3a13.md` `init_resource` rule):

| Probe | Outcome | baselineMatches | inScopeMatches |
|---|---|---|---|
| With manifest in corpus | `out-of-scope` | 1 (`.totem/compiled-rules.json`) | 3 (legitimate production hits) |
| Without manifest in corpus | `candidate-debt` | 0 | 3 (legitimate production hits) |

The 1.19.0 ship of ADR-091 Stage 4 (`mmnto-ai/totem#1682`) ran T1 tests against synthetic rule corpora, not against rules co-located in a real `.totem/compiled-rules.json` manifest, so this didn't surface in CR review.

## Probe script (repro)

`D:\dev\totem\scripts\stage4-probe-1761.mjs` carries a working repro. `EXCLUDE_MANIFEST=1` env switch toggles the manifest in/out of the corpus. Working tree at `mmnto-ai/liquid-city` HEAD (post-`#135` totem 1.19.0 bump).

## Proposed fix

The Stage 4 verifier should skip the manifest file from the corpus before running the rule. Either:

1. **Hardcode in the verifier:** Add `.totem/compiled-rules.json` (and any sibling manifests like `.totem/lessons.md`, `.totem/lessons/*.md`) to a verifier-internal exclusion list applied inside `verifyAgainstCodebase` before passing files to `runRuleAgainstAllFiles`. Cheapest, no API surface change.
2. **Add to `DEFAULT_BASELINE_GLOBS`:** Append `**/.totem/**` to the default baseline. Symmetric with how `tests/`/`fixtures/` are handled. Slight semantic difference: with this approach, hits in `.totem/` would count as baseline matches (still archives the rule) rather than being skipped entirely.
3. **Filter at `listFiles` callback:** Push the responsibility to the CLI integration site. Less general — every Stage 4 caller would need to remember the filter.

Lean: option 1 (hardcoded exclusion in the verifier). Cleanest semantics: the manifest is verifier-internal data, not source code, and treating it as either source or baseline is a category error. Option 2 is acceptable as the second-best, and is consistent with how `T2` (`mmnto-ai/totem#1683`) layers consumer baseline overrides.

## Acceptance criteria

- [ ] `verifyAgainstCodebase` skips `.totem/compiled-rules.json` from the corpus before classification.
- [ ] Whether other manifest paths (e.g., `.totem/lessons/*.md`, `.totem/lessons.md`) should also be skipped — surface as Open Question. Lessons are the source of `badExample` text after Pipeline 1/2/3 compile, so they have the same self-match risk.
- [ ] Test fixture: a regex rule whose `badExample` appears in a synthetic `compiled-rules.json` placed inside the test working tree → outcome routes correctly without the self-match.
- [ ] Test fixture: a regex rule whose pattern legitimately matches a non-manifest file (production code) → outcome unchanged (still archives or routes per `classifyFile`).

## Related

- ADR-091 Stage 4 cluster: `mmnto-ai/totem#1682` (T1, shipped 1.19.0), `mmnto-ai/totem#1683` (T2 baseline overrides — natural carrier).
- `mmnto-ai/totem#1761` AC #1 probe (this finding).
- `mmnto-ai/totem#1758` (matchesGlob consolidation, also slated for `#1683` ride-along).

## Tier

Tier-2. Behavioral correctness gap in load-bearing substrate. The substrate quirk currently *masks* over-broad rules — when a rule legitimately needs Stage 4 to detect over-broad firing, the manifest self-match short-circuits to `out-of-scope` before the in-scope hits get a chance to route to `candidate-debt`. The result is a false sense of "calibration works" that doesn't reflect real codebase risk. Fix natural carrier is `mmnto-ai/totem#1683` (PR3).

### PR DESCRIPTION
**Pull request by satur8d**
## Summary

Bundles three substrate fixes that all touch `packages/core/src/stage4-verifier.ts` overlapping surfaces:

- **`mmnto-ai/totem#1683`** (T2 baseline overrides) — `review.stage4Baseline` config + `# stage4-baseline:` `.totemignore` directives + `resolveStage4Baseline` resolver + naming-discipline guard rejecting `allowlist` keys.
- **`mmnto-ai/totem#1758`** (matchesGlob consolidation) — Stage 4's local regex-based matcher had a substring hole (`**/tests/**` matched `src/contests/foo.ts`); consolidated onto pattern-specific `fileMatchesGlobs` from `rule-engine.ts`.
- **`mmnto-ai/totem#1765`** (manifest self-match exclusion) — `STAGE4_MANIFEST_EXCLUSIONS` filter at the CLI integration boundary, demonstrated cleanly by the `mmnto-ai/totem#1761` AC #1 probe.

Pre-approved bundling rationale + 4 open questions in the [design doc](.totem/specs/1683.md).

## Scope expansion mid-implementation

Consolidating onto rule-engine's `matchesGlob` surfaced a separate latent bug: `**/<rest>` recursion stripped `**/` and required `<rest>` to match at path-root, so `**/__tests__/**` failed on `packages/cli/src/__tests__/foo.ts`. Existing rule-engine tests didn't cover this shape (no compiled rules use it). Fixed by walking every `/`-aligned tail of the path during `**/` recursion. **3 new regression tests on the rule-engine matcher.**

## Deviation from approved design Q4

Held `postmerge/1743-1747-1745-1749` ride-along local instead of bundling. The diff turned out to be 41 lesson files + 336 lines of `compiled-rules.json` + 182 lines of auto-mirror docs — review surface that bots would inspect, not "metadata-only." Will ride a future feature PR.

## Public API additions (`@mmnto/totem`)

- `resolveStage4Baseline(input: ResolveStage4BaselineInput): Stage4Baseline` — pure resolver. Composition: `defaults ∪ ignoreDirectives ∪ configExtend ∖ configExclude`.
- `parseStage4BaselineDirectives(content: string): string[]` — pure parser for `# stage4-baseline: <glob>` lines.
- `STAGE4_MANIFEST_EXCLUSIONS: readonly string[]` — `['.totem/compiled-rules.json']`.
- `fileMatchesGlobs(filePath, globs)` — promoted from file-local to exported (single source of truth for glob semantics across rule-engine and Stage 4).
- `Stage4Baseline` interface extended with provenance fields (`extendedFromIgnoreFile`, `extendedFromConfig`, `excludedFromConfig`) for `totem doctor` (T4) UX.
- `ReviewConfigSchema.stage4Baseline` Zod field with `superRefine` rejecting `allowlist` keys.

## Sonnet pre-push tail

R0 caught a real CRITICAL: `fileMatchesGlobs` has default-allow semantics on empty positive sets (returns `true` so a rule with empty `fileGlobs` runs everywhere — that's the rule-engine contract). Stage 4's baseline doesn't share that contract — empty `excludeFileGlobs` means "nothing is baseline-excluded," not "everything is." Without an explicit length guard, a consumer who `exclude`-s every default would route every file to baseline classification → all rule violations suppressed. Added length guard + dedicated regression test in `verifyAgainstCodebase with consumer excluding every default baseline glob`.

## Test counts

- Core: 1612 → 1651 (+39 — `resolveStage4Baseline`, `parseStage4BaselineDirectives`, `STAGE4_MANIFEST_EXCLUSIONS`, `**/dir/**` regression, empty-baseline guard, schema validation, naming-discipline guard).
- CLI: 1958 (unchanged — wiring exercised via core's stage4-verifier integration tests).
- Total: 3609/3609 passing.

## Test plan

- [x] `pnpm --filter @mmnto/totem build` — clean.
- [x] `pnpm --filter @mmnto/cli build` — clean.
- [x] Full test suites — 3609 pass.
- [x] `pnpm exec totem lint` — PASS (22 warnings, all false-positive class on `toContain` array assertions matching the secret-masking lesson regex, and existing-line lints not from this PR).
- [x] `pnpm exec totem review` — PASS (R0 CRITICAL applied + new regression test).
- [x] Smoke: re-run `mmnto-ai/totem#1761` AC #1 probe with this branch built — confirm rule routes to `candidate-debt` (not `out-of-scope`) when manifest is excluded from corpus. (Validated via `EXCLUDE_MANIFEST=1` mode of the original probe; substrate fix matches the simulated outcome.)
- [ ] CR + GCA review.
- [ ] Smoke after merge: run `totem lesson compile` on totem itself with the bundle live and confirm no regression on the existing 452-rule manifest.

Closes #1683
Closes #1758
Closes #1765

### CODE REVIEW

**Comment by satur8d** [comment]
**R1 amendment — 3 findings applied (`b9edd513`):**

- **GCA HIGH (`stage4-verifier.ts:258` — negative-only baseline still default-allows):** Right call. The R0 `length > 0` guard caught the fully-empty case but missed the negative-only edge. A consumer who passes `extend: ['!**/foo']` AND excludes every default would produce `excludeFileGlobs = ['!**/foo']` — length 1, zero positives → `fileMatchesGlobs` falls into default-allow → every file misclassified as baseline. Replaced with the suggested `hasPositiveBaseline` check. Regression test: synthetic baseline with only `!`-prefixed entries asserts in-scope hits survive.

- **CR MAJOR (`config-schema.ts:250` — `.passthrough()` accepts typos):** Right call. `exlcude`/`extends`/`excludes` would have parsed cleanly and silently been ignored by the resolver. Added an explicit unknown-keys check in the `superRefine` using a `STAGE4_BASELINE_KNOWN_KEYS = {'extend', 'exclude'}` set. The custom `allowlist` error message (with the `mmnto-ai/totem#1683` pointer) is preserved — still surfaces ahead of the generic unknown-key issue when both apply. Three regression tests cover the typo case, the plural case, and the mixed-with-allowlist case.

- **CR MINOR (`config-reference.md:43` — sample mixes `reviewOptions` and `review`):** Right call. `reviewOptions` was inert (not in `TotemConfigSchema`); the example was misleading because copied configs would have the older key as a no-op alongside the new `review.stage4Baseline`. Replaced the inert block with a comment block noting `TotemConfigSchema` reads only `review`. Single key, no copy-paste trap.

Pre-push: format clean, `totem lint` PASS (26 warnings, all false-positive class), Sonnet review PASS.

Test counts: core 1651 → 1655 (+4 R1 regression tests).

@coderabbitai full review

File: packages/cli/src/commands/compile.ts [hunk 1]
```diff
@@ -672,9 +686,33 @@ export async function compileCommand(
...
+        // not a silent-skip case).
+        const ignorePath = path.join(repoRoot, '.totemignore');
+        let ignoreContent = '';
+        try {
+          ignoreContent = await fs.promises.readFile(ignorePath, 'utf-8');
+        } catch (err) {
+          const code = (err as NodeJS.ErrnoException).code;
+          if (code !== 'ENOENT') throw err;
+        }
+        const ignoreDirectives = parseStage4BaselineDirectives(ignoreContent);
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/commands/compile.ts:716]: False positive — the parser already requires the `#` prefix. The regex at `packages/core/src/stage4-verifier.ts` is:

```ts
const STAGE4_BASELINE_DIRECTIVE_RE = /^#\s*stage4-baseline:\s*(.+?)\s*$/;
```

The `^#` anchor is load-bearing — only lines starting with `#` (with optional whitespace) match. Tests at `packages/core/src/stage4-verifier.test.ts:67-107` cover the documented `# stage4-baseline: <glob>` form with variable whitespace (`#stage4-baseline:foo`, `# stage4-baseline: foo`, `#  stage4-baseline:  foo  `, CRLF, leading-`#` case-sensitivity) — all currently passing.

Wired call site in `compile.ts` passes the raw `.totemignore` content with `#` prefixes intact. The suggested fix would actually break correctness — preprocessing to strip `#` would make every line subsequently fail the `^#` anchor.

No code change. Closing this thread.

**Comment by satur8d** [comment]
**R2 amendment — 4 applied + 1 defended (`f53c103e`):**

**Applied:**

- **CR MAJOR (`compile.ts:658` — `config.totemDir` override unhandled):** Right call. Hardcoded `STAGE4_MANIFEST_EXCLUSIONS = ['.totem/compiled-rules.json']` would miss the active manifest for consumers who override `config.totemDir`. Computed `path.join(config.totemDir, 'compiled-rules.json').replace(/\/g, '/')` and added to the exclusion set alongside the default. The same self-match self-archive bug from `mmnto-ai/totem#1765` would have returned for those consumers without this fix.

- **CR OUTSIDE-DIFF MAJOR (`rule-engine.ts:84` — literal-path overmatch):** Right call. This is the SECOND hole `mmnto-ai/totem#1758` explicitly called out: `endsWith('/' + glob)` caused `matchesGlob('packages/src/foo.ts', 'src/foo.ts')` to return `true`. With Stage 4 now sourced from `fileMatchesGlobs`, the bug became load-bearing for both rule-engine consumers AND Stage 4 baseline classification. Fix: when `glob.includes('/')`, require exact normalized-path match. Bare-filename globs (no `/`) keep the original suffix-match behavior so `Dockerfile` still matches `src/Dockerfile`. Two regression tests on the rule-engine matcher.

- **CR OUTSIDE-DIFF MINOR (`.changeset` — PR housekeeping in changelog):** Right call. Removed the "Bot-review tail" paragraph. Changeset stays focused on user-facing release notes; PR-internal review process belongs in the PR description, not the published changelog.

- **CR NITPICK (`stage4-verifier.ts:185-218` — TSDoc on public helpers):** Restored the TSDoc blocks on `resolveStage4Baseline`, `parseStage4BaselineDirectives`, and `ResolveStage4BaselineInput`'s fields. These got dropped during the original implementation when an edit hook split the module helpers into smaller pieces. IDE hovers and generated API docs now carry the contract.

**Defended:**

- **CR MAJOR (`compile.ts:707` — claim: parser drops `#` prefix):** False positive — already replied on the inline thread. The parser regex at `stage4-verifier.ts` is `/^#\s*stage4-baseline:\s*(.+?)\s*$/`. The `^#` anchor is load-bearing; existing tests at `stage4-verifier.test.ts:67-107` cover the `#`-prefixed form with variable whitespace. The suggested fix would actually break correctness — preprocessing to strip `#` would make every line subsequently fail the `^#` anchor.

**Pre-push tail:**

The literal-path glob fix tripped a totem-lint rule (lesson `6c0f60f3` flags `.includes('/')` patterns and asks for `.includes('/', 1)`). False positive in this case — the rule's `,1` start-index is meant to skip a leading slash for path-prefix detection, but my use case is "any slash" detection (path-shape vs bare-filename distinction). Added a `// totem-context:` directive citing `mmnto-ai/totem#1758` per the discipline rule.

Pre-push: format clean, `totem lint` PASS (0 violations), Sonnet review PASS.

Test counts: core 1655 → 1657 (+2 R2 regression tests for the path-shaped literal-glob fix).

@coderabbitai full review

**Comment by satur8d** [comment]
**R3 amendment — 1 applied (`65a8b4b7`):**

**CR MAJOR (`stage4-verifier.ts:291-313` — subtract rule-declared scope from baseline before classifying):** Confirmed. Real bug. A test-contract rule with `fileGlobs: ['**/*.test.*']` running against `src/foo.test.ts` matched rule scope (step 1, continue), then re-matched the same file against `baseline.excludeFileGlobs` (which contains `**/*.test.*` from `DEFAULT_BASELINE_GLOBS`), returned `'baseline'`, and flipped the outcome to `out-of-scope` — archiving the rule on its own intended corpus. Applied your suggested diff verbatim:

```ts
const effectiveBaselineGlobs =
  ruleFileGlobs && ruleFileGlobs.length > 0
    ? baseline.excludeFileGlobs.filter((g) => !ruleFileGlobs.includes(g))
    : baseline.excludeFileGlobs;
```

Byte-equal subtraction is consistent with `resolveStage4Baseline`'s `exclude` semantics — an explicit `fileGlobs` entry matching a baseline entry is the rule's way of claiming that part of the baseline as its own. Subtraction is surgical: non-claimed baseline globs still apply.

**Three regression tests** added in `stage4-verifier.test.ts` under a new `verifyAgainstCodebase test-contract rule scope vs baseline (CR mmnto-ai/totem#1766 R3)` describe block:

1. **Test-contract scope:** rule scoped to `**/*.test.*`, file `foo.test.ts`, hit produces `in-scope-bad-example` (not `out-of-scope`).
2. **Fixture-contract scope:** rule scoped to `**/__fixtures__/**`, file `__fixtures__/sample.ts`, hit produces `in-scope-bad-example`.
3. **No over-subtraction:** rule scoped to `**/*.test.*` only, file `packages/__tests__/foo.test.ts` (matches both rule scope AND the non-claimed `**/__tests__/**` baseline glob) — still classifies as baseline. Locks in surgical subtraction.

Doc comments updated to enumerate the three empty-positive cases (Sonnet R0 + GCA R1 + new R3 case where rule `fileGlobs` covers every positive baseline entry).

Pre-push: `pnpm --filter @mmnto/totem test stage4-verifier.test` → 44/44 pass (3 new + 41 prior). `totem lint` → PASS (warnings only, all false positives on async-test-callback heuristic + `.includes()` heuristic — `verifyAgainstCodebase` is genuinely async; `Array.includes` is byte-equal subtraction not filename-substring matching). `totem review` → PASS, no findings.

This is a real correctness fix for the test-contract scope class — the substrate already passed multi-round review through `f53c103e`, but classifyFile had this latent bug for any rule whose `fileGlobs` overlapped a baseline entry. With T3/T4 (`#1684`/`#1685`) ahead, a rule that legitimately targets test files would have been silently archived in Stage 4.

@coderabbitai full review

**Comment by satur8d** [comment]
**R4 amendment — 1 nit applied + 1 strict-subset edge case captured (`f7af6f5c`):**

**Applied — CR Nit (Low value):** Added TSDoc on `STAGE4_MANIFEST_EXCLUSIONS` (`packages/core/src/stage4-verifier.ts:173`) noting the CLI integration site's dynamic `totemDir`-aware companion path computation. Constant + runtime-computed path together cover both default and overridden installs.

**Captured — strict-subset rule scope vs baseline glob (CR R3 learning):** Your R3 review note that `Array.includes` is byte-equal (not glob-subset aware) is the right intentional contract — an explicit `fileGlobs` entry matching a baseline entry is the rule's way of claiming that part of the baseline as its own. Glob-subset resolution would invert that: a narrow rule glob would silently subtract a broader baseline glob without explicit author intent. But the failure mode for a strict-subset rule (`fileGlobs: ['**/*.test.ts']` vs default baseline `**/*.test.*`) IS silent — the rule gets archived on its own intended corpus with no UX signal.

Captured in two places per the right-tracker disposition:

1. **`.totem/specs/1683.md`** — new "Known limitations (deferred to T3/T4)" section documenting the case + the workarounds today (rule author byte-equal claim, or consumer `review.stage4Baseline.exclude`).
2. **Cross-cutting tracking comment posted on `mmnto-ai/totem#1685`** ([link](https://github.com/mmnto-ai/totem/issues/1685#issuecomment-4356872868)) since T4 doctor UX is the natural surface to flag the misclassification class to humans. Cross-references `mmnto-ai/totem#1684` (T3 metrics) for the field-corpus signal that would feed compile-worker prompt iteration if the metric is non-zero.

**Non-blocking pre-T3/T4** — confirmed corpus today shows test-contract rules emit `**/*.test.*` (byte-equal form) more often than `**/*.test.ts` (strict-subset form). LC #135's `init_resource` rule (the original AC #1 probe target) ships with `**/*.ts` — no overlap with default baseline at all, so isn't in this class.

Pre-push: `pnpm --filter @mmnto/totem build` PASS, `totem lint` PASS (383 rules, 0 violations after shortening the new spec heading to ≤60 chars to satisfy the lesson-heading rule), `totem review` PASS, no findings.

@coderabbitai full review
